modules/Bio/EnsEMBL/Hive/DBSQL/AnalysisStatsAdaptor.pm

```diff
@@ -193,5 +193,18 @@ sub increment_a_counter {
     }
 }
 
+
+sub get_seconds_since_locked {
+    my ($self, $analysis_id) = @_;
+
+    my $sql = 'SELECT '  . $self->dbc->_interval_seconds_sql('MAX(when_logged)') . ' FROM analysis_stats_monitor WHERE analysis_id = ? AND sync_lock = 0';
+    my $sth = $self->prepare($sql);
+    $sth->execute($analysis_id);
+    my ($seconds_since_locked) = $sth->fetchrow_array();
+    $sth->finish;
+    return $seconds_since_locked;
+}
+
+
```

Comment:
The same, do we need process case when request is failed?

Reply:
The query may return `NULL` indeed. I'll see if I handle this case here or where this method is called (in Queen).